# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeParent.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 gfx/layers/ipc/CompositorBridgeParent.h
--- a/gfx/layers/ipc/CompositorBridgeParent.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/gfx/layers/ipc/CompositorBridgeParent.h	Wed Jan 08 22:19:16 2020 +0000
@@ -56,7 +56,6 @@
 
 namespace dom {
 class HostWebGLCommandSink;
-class HostWebGLErrorSource;
 class WebGLParent;
 }  // namespace dom
 
@@ -304,15 +303,8 @@
     return IPC_FAIL_NO_REASON(this);
   }
 
-  virtual PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
-      const HostWebGLErrorSource& aErrorSource) = 0;
-
-  virtual PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
-      HostWebGLErrorSource&& aErrorSource) = 0;
-
-  virtual bool DeallocPWebGLParent(PWebGLParent* aWebGLParent) = 0;
+  virtual already_AddRefed<PWebGLParent> AllocPWebGLParent(
+      const webgl::InitContextDesc&, webgl::InitContextResult* out) = 0;
 
   bool mCanSend;
 
@@ -702,28 +694,13 @@
   WebRenderBridgeParent* GetWrBridge() { return mWrBridge; }
   webgpu::WebGPUParent* GetWebGPUBridge() { return mWebGPUBridge; }
 
-  PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, const HostWebGLCommandSink& aCommandSink,
-      const HostWebGLErrorSource& aErrorSource) override {
+  already_AddRefed<PWebGLParent> AllocPWebGLParent(
+      const webgl::InitContextDesc&, webgl::InitContextResult*) override {
     MOZ_ASSERT_UNREACHABLE(
         "This message is CrossProcessCompositorBridgeParent only");
     return nullptr;
   }
 
-  PWebGLParent* AllocPWebGLParent(
-      const WebGLVersion& aVersion, HostWebGLCommandSink&& aCommandSink,
-      HostWebGLErrorSource&& aErrorSource) override {
-    MOZ_ASSERT_UNREACHABLE(
-        "This message is CrossProcessCompositorBridgeParent only");
-    return nullptr;
-  }
-
-  bool DeallocPWebGLParent(PWebGLParent* aWebGLParent) override {
-    MOZ_ASSERT_UNREACHABLE(
-        "This message is CrossProcessCompositorBridgeParent only");
-    return false;
-  }
-
  private:
   void Initialize();
 